src/main/java/org/gnuhpc/bigdata/service/CollectorService.java

```diff
@@ -209,23 +208,29 @@ private void getMatchingAttributes(MBeanServerConnection mBeanServerConnection,
   public HashMap<String, Object> listJMXFilterTemplate(String filterKey) {
     HashMap<String, Object> filterTemplateMap = new HashMap<>();
     HashMap<Object, Object> yamlHash;
+    String projectRootPath = "";
     try {
-      File tempDir = new ClassPathResource("JMXFilterTemplate/").getFile();
-      if (tempDir.isDirectory()) {
-        for (File yamlFile:tempDir.listFiles()) {
-          String fileFullName = yamlFile.getName();
-          if (matchIgnoreCase(filterKey, fileFullName)) {
-            String[] fileNames = fileFullName.split("\\.");
-            FileInputStream yamlInputStream = new FileInputStream(yamlFile);
-            yamlHash = (HashMap<Object, Object>) new Yaml().load(yamlInputStream);
-            filterTemplateMap.put(fileNames[0], yamlHash);
-          }
+      File tempFile = new File("");
+      projectRootPath = tempFile.getCanonicalPath();
+      log.info("Project Root Path is " + projectRootPath);
+      File jmxFilterDir = new File(projectRootPath +"/JMXFilterTemplate");
+      if (!jmxFilterDir.exists() || !jmxFilterDir.isDirectory()) {
+        throw new IOException();
+      }
+      for (File yamlFile:jmxFilterDir.listFiles()) {
+        String fileFullName = yamlFile.getName();
+        log.info("Found JMXFilterTemplate filename=" + fileFullName);
+        if (matchIgnoreCase(filterKey, fileFullName)) {
```

Comment:
please use jackson-dataformat-yaml to parse yaml config file if it can satisfy your requirements.